uwblueprint/marillac-place
# Backend refactor

## Notion ticket link


- [Resident refactor](https://www.notion.so/uwblueprintexecs/Refactor-resident-service-bb85418874554227a1bbf66f4b87e7c2?pvs=4) - https://github.com/uwblueprint/marillac-place/pull/51/files
- [Staff refactor](https://www.notion.so/uwblueprintexecs/Refactor-staff-service-c97fd65cef614ee3836423e6daa617ef?pvs=4) - https://github.com/uwblueprint/marillac-place/pull/51/files
- [Warning refactor](https://www.notion.so/uwblueprintexecs/Refactor-warning-service-functionality-2072a3589436461a8a6cfedd98819df8?pvs=4) - https://github.com/uwblueprint/marillac-place/pull/36/files
- [Notifications refactor](https://www.notion.so/uwblueprintexecs/Refactor-notifications-service-functionality-854f2a1cb86f431c9ed2f52d323859f7?pvs=4) - https://github.com/uwblueprint/marillac-place/pull/53/files
- [Backend file organization](https://www.notion.so/uwblueprintexecs/Remove-python-directory-and-move-files-out-of-typescript-directory-ac06b2032d304d65bc7f977cba7d8c7a?pvs=4)


## Implementation description

* Merged the data model refactor along with all refactoring work
* Fixed errors that resulted from the merge
* Cleaned up some of the DTOs
* Renamed some of the queries and mutations for consistency
* Added a `userType` parameter to the `login` mutation
* Removed the `backend/python` and `backend/typescript` directories, and moved all backend code under `backend/` directly
* Updated GH workflows based on the above directory change 
* Changed the name of the Docker containers from `scv2` to `mp`
* Added the required Prisma migrations to the local file system and removed the starter code Sequelize migrations
* Removed unused starter code


## Steps to test

1. Delete your old Marilac Place images and containers from Docker Desktop
2. Pull the code from this branch by running `git fetch && git checkout backend-refactor`
3. Update your `.env` and `/backend/.env` files
4. Run the codebase by running `docker compose up --build`. Ensure that the db, backend, and frontend containers run without error
2. Test all queries and mutations that you were responsible for to ensure that functionality is still working


## What should reviewers focus on?

## Checklist
